cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -947,10 +949,31 @@ fn insert_common_epilogue(
     }
 }
 
-pub fn emit_unwind_info(func: &ir::Function, isa: &dyn TargetIsa, mem: &mut Vec<u8>) {
-    // Assumption: RBP is being used as the frame pointer
-    // In the future, Windows fastcall codegen should usually omit the frame pointer
-    if let Some(info) = UnwindInfo::try_from_func(func, isa, Some(RU::rbp.into())) {
-        info.emit(mem);
+pub fn emit_unwind_info(
+    func: &ir::Function,
+    isa: &dyn TargetIsa,
+    kind: FrameUnwindKind,
+    sink: &mut dyn FrameUnwindSink,
+) {
+    match kind {
+        FrameUnwindKind::Fastcall => {
+            let mut mem = Vec::new();
+            // Assumption: RBP is being used as the frame pointer
+            // In the future, Windows fastcall codegen should usually omit the frame pointer
+            if let Some(info) = UnwindInfo::try_from_func(func, isa, Some(RU::rbp.into())) {
+                info.emit(&mut mem);
```

Comment:
I think we'll want to pass the sink on down into `UnwindInfo::emit` so we're not double buffering these bytes (once here and then presumably in the caller's sink impl).

Reply:
Also, I'm totally open to renaming `UnwindInfo` so it's more specific to Windows so this gets less confusing.

Reply:
Not an issue atm.